dotcom-rendering/src/components/WeatherWrapper.importable.tsx

```diff
@@ -6,7 +6,8 @@ import { Weather, WeatherPlaceholder } from './Weather';
 const appendPartnerCodeToUrl = (
 	url: string | undefined,
 ): string | undefined => {
-	if (!url || !URL.canParse(url)) {
+	// eslint-disable-next-line @typescript-eslint/strict-boolean-expressions, @typescript-eslint/no-unnecessary-condition -- We're on the client so we don't know if URL.canParse is available
+	if (!url || (!!window.URL.canParse && !URL.canParse(url))) {
```

Comment:
If I understand correctly, the purpose of this condition is to check whether the URL is parseable, and return `undefined` if it's not?

If `canParse` is undefined this whole condition evaluates to `false`. If this condition evaluates to `false`, we skip to the next block where the URL is parsed, which **may crash** because we skipped the check to see if it's parseable?

Rather than trying to handle this case too, perhaps this function would be simpler rewritten as a `try...catch` wrapped around the `URL` constructor?